pkg/operator/configobservation/auth/podsecurityadmission_test.go

```diff
@@ -25,78 +26,62 @@ func TestObservePodSecurityAdmissionEnforcement(t *testing.T) {
 	restrictedJSON, err := json.Marshal(restrictedMap)
 	require.NoError(t, err)
 
-	defaultFeatureSet := &configv1.FeatureGate{
-		Spec: configv1.FeatureGateSpec{
-			FeatureGateSelection: configv1.FeatureGateSelection{
-				FeatureSet:      "",
-				CustomNoUpgrade: nil,
-			},
-		},
-	}
+	defaultFeatureSet := featuregates.NewHardcodedFeatureGateAccess([]configv1.FeatureGateName{}, []configv1.FeatureGateName{})
 
-	corruptFeatureSet := &configv1.FeatureGate{
-		Spec: configv1.FeatureGateSpec{
-			FeatureGateSelection: configv1.FeatureGateSelection{
-				FeatureSet:      "Bad",
-				CustomNoUpgrade: nil,
-			},
-		},
-	}
+	// TODO provide a hardcoded test harness that allows not-ready and error
```

Comment:
I see that hardcodedFeatureGateAccess supports this but doesn't export any way to configure it. Are you imagining that it should be made usable for external tests like these? I don't object to dropping a stub implementation right here for use in not ready and error test cases.